### PR TITLE
autoprune minor post-merge improvements

### PR DESCRIPTION
**Pull request by jonasschnelli**
- Adds information about whether we are in pruned mode or not to RPC `getblockchaininfo()`.
- Produces appropriate response when trying to access a pruned block over REST and RPC.

### CODE REVIEW

**Comment by sipa** [comment]
Please don't extend getinfo. We want to deprecate it.

**Comment by jonasschnelli** [comment]
@sipa: right. You already told me that on IRC. My mistake, sorry.
Moved information to `getblockchaininfo()`.

File: src/rpcblockchain.cpp [hunk 1]
```diff
@@ -492,6 +492,15 @@ Value getblockchaininfo(const Array& params, bool fHelp)
     obj.push_back(Pair("difficulty",            (double)GetDifficulty()));
     obj.push_back(Pair("verificationprogress",  Checkpoints::GuessVerificationProgress(chainActive.Tip())));
     obj.push_back(Pair("chainwork",             chainActive.Tip()->nChainWork.GetHex()));
+    obj.push_back(Pair("autoprune",             fPruneMode));
```
Comment (sipa) [inline src/rpcblockchain.cpp]: I think we're abolishing the name "autoprune". Can you just call it "pruned" or something?

**Comment by jgarzik** [comment]
ut ACK

**Comment by jonasschnelli** [comment]
Renamed and added some error handling for REST and RPC when trying to access a pruned block.

**Comment by sipa** [comment]
utACK

**Comment by laanwj** [comment]
utACK after @sdaftuar 's nit fixed. The error message specifically mentions "pruned", so it should be sure that's actually the case.

**Comment by jonasschnelli** [comment]
Agreed with @sdaftuar.
Corrected prune mode test.

**Pull request merged**